src/main/java/org/broadinstitute/dsm/util/externalShipper/GBFRequestUtil.java

```diff
@@ -207,13 +256,57 @@ public void orderStatus(ArrayList<KitRequest> kitRequests) throws Exception {
         }
     }
 
+
+    public static void updateDeliveredDateForKit(KitRequest kit) {// for when UPS is integrated
+        String query = "UPDATE ddp_kit SET delivered_date= ? where dsm_kit_id <> 0 and dsm_kit_id in (SELECT kit.dsm_kit_id from (Select * from ddp_kit) kit  where kit.dsm_kit_request_id  = ? )";
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            dbVals.resultValue = null;
+            try (PreparedStatement stmt = conn.prepareStatement(query)) {
+                stmt.setLong(1, System.currentTimeMillis() / 1000);
+                stmt.setString(2, kit.getDsmKitRequestId());
+                int result = stmt.executeUpdate();
+            }
+            catch (Exception e) {
+                dbVals.resultException = e;
+            }
+            return dbVals;
+        });
+
+        if (results.resultException != null) {
+            throw new RuntimeException("Error getQueryDetail ", results.resultException);
+        }
+    }
+
+    public static void updateReceivedDateForKit(KitRequest kit) {// for when UPS is integrated
```

Comment:
method does same as updateDeliveredDateForKit. just make one method and pass in the different query from where you call that method!!!

Reply:
not called method, in here only for ups. Will optimize later